src/command.c

```diff
@@ -499,63 +499,74 @@ static bool cmd_rtt(target *t, int argc, const char **argv)
 {
 	(void)t;
 	const size_t command_len = strlen(argv[1]);
-	if (argc == 1 || (argc == 2 && !strncmp(argv[1], "enabled", command_len))) {
+	if (argc == 1 || (argc == 2 && strncmp(argv[1], "enabled", command_len) == 0)) {
 		rtt_enabled = true;
 		rtt_found = false;
-	} else if ((argc == 2) && !strncmp(argv[1], "disabled", command_len)) {
+		memset(rtt_channel, 0, sizeof(rtt_channel));
+	} else if ((argc == 2) && strncmp(argv[1], "disabled", command_len) == 0) {
```

Comment:
Please remove the extra parentheses from around the argc comparison here, and in the "status" block